service/storage_service.cc

```diff
@@ -2413,7 +2413,12 @@ future<> topology_coordinator::fence_previous_coordinator() noexcept {
         } catch (...) {
             slogger.error("raft topology: failed to fence previous coordinator {}", std::current_exception());
         }
-        co_await seastar::sleep_abortable(std::chrono::seconds(1), _as);
+        try {
+            co_await seastar::sleep_abortable(std::chrono::seconds(1), _as);
+        } catch (abort_requested_exception&) {
+            // Abort was requested. Break the loop
+            break;
+        }
```

Comment:
What if it throws something else?

Reply:
If it throws something unexpected, e.g. `std::bad_alloc` then I think it's fine to abort as the situation is already bad or something is broken inside `sleep_abortable`. This is how it is now handled in v2.

The only alternative I see is to continue the loop, but then we risk loop spinning if `sleep_abortable` is broken, so I'm not sure this is fine.

Reply:
I guess if we do want `topology_coordinator::run` to handle all the exceptions internally and never cause a crash then we should `continue` here. It's how it is already done in `cdc_generation_publisher_fiber`. I'll adjust the code here for consistency.